plugins/forc-index/src/ops/forc_index_start.rs

```diff
@@ -19,9 +21,37 @@ pub fn init(command: StartCommand) -> anyhow::Result<()> {
         run_migrations,
         metrics,
         manifest,
+        auto_setup_database,
         ..
     } = command;
 
+    if auto_setup_database {
+        let name = postgres_database
+            .clone()
+            .unwrap_or(defaults::POSTGRES_DATABASE.to_string());
+        let password = postgres_password
+            .clone()
+            .unwrap_or(defaults::POSTGRES_PASSWORD.to_string());
+        let user = postgres_user
+            .clone()
+            .unwrap_or(defaults::POSTGRES_USER.to_string());
+        let port = postgres_port
+            .clone()
+            .unwrap_or(defaults::POSTGRES_PORT.to_string());
+
+        let create_db_cmd = CreateDbCommand {
```

Comment:
@deekerno 
- The trickier part (not really) here is, if the user already has this database setup, we only want to to start the database - not set it up again.
- @rvmelkonian I agree you can stub your toe on the "didn't know I already had a DB running" but I do think that should be the responsibility of the dev to know this
  - We could add some warning or something to let them know if something is already running on that port (but that's outside the scope of this PR - feel free to create an issue for this!)

Reply:
@ra0x3 Added a bit to only start if the database already exists. I had to do a string comparison as I couldn't get a downcast to the innermost error to properly work; it's just as well since doing so would have required me to pull in `sqlx-core` as a dependency.